src/physics/types.rs

```diff
@@ -142,6 +142,26 @@ impl Field {
     }
 }
 
+
+// Invironment ///////////////////////////////////////////////////////////////
+
+pub struct Invironment {
+
+    pub bodies: HashMap<u32, Body>,
+    pub fields: HashMap<u32, Field>,
```

Comment:
I would suggest to store the fields in a `Vector`, rather than a `HashMap`, there's no reason to keep in in a `HashMap`.

Reply:
how about this : pub fields: Vec<Body>, ?